executor/common_linux.h

```diff
@@ -2393,14 +2393,13 @@ struct fs_image_segment {
 #define sys_memfd_create 279
 #endif
 
-static unsigned long fs_image_segment_check(unsigned long size, unsigned long nsegs, long segments)
+static unsigned long fs_image_segment_check(unsigned long size, unsigned long nsegs, struct fs_image_segment* segs)
 {
 	unsigned long i;
 	// Strictly saying we ought to do a nonfailing copyout of segments into a local var.
 	// But some filesystems have large number of segments (2000+),
 	// we can't allocate that much on stack and allocating elsewhere is problematic,
 	// so we just use the memory allocated by fuzzer.
-	struct fs_image_segment* segs = (struct fs_image_segment*)segments;
```

Comment:
While we are here: please also remove the comment above.
It's unrelated to your change, but we don't use NONFAILING in pseudo-syscalls at all, so it's not relevant anymore.